xarray/backends/zarr.py

```diff
@@ -676,7 +676,7 @@ def set_variables(self, variables, check_encoding_set, writer, unlimited_dims=No
                 # and append_dim.
                 if self._write_empty is not None:
                     zarr_array = zarr.open(
-                        store=self.zarr_group.store,
+                        store=self.zarr_group.chunk_store,
```

Comment:
Do you understand why the `store` interface is read-only and the `chunk_store` is not? I'm curious if this happens for all stores or only the directory store?

Reply:
Thanks for this excellent explanation.

I believe we created a bit of a mess with consolidated metadata. Given the way things work in Zarr, I think this PR is an acceptable solution.

My one suggestion would be to put ☝️ that super useful explanation as an inline comment right in the code. It will be extremely helpful for future maintainers.